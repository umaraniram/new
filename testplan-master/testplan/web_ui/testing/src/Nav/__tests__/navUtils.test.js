import React from 'react';
import {StyleSheetTestUtils} from "aphrodite";

import {CreateNavButtons, GetSelectedUid} from '../navUtils';
import {TESTPLAN_REPORT} from '../../Common/sampleReports';

describe('navUtils', () => {

  describe('CreateNavButtons', () => {

    beforeEach(() => {
      // Stop Aphrodite from injecting styles, this crashes the tests.
      StyleSheetTestUtils.suppressStyleInjection();
    });

    afterEach(() => {
      // Resume style injection once test is finished.
      StyleSheetTestUtils.clearBufferAndResumeStyleInjection();
    });

    it('returns an array of nav buttons', () => {
      const props = {
        breadcrumbLength: 1,
        displayTags: false,
        displayTime: false,
        displayEmpty: true,
        handleNavClick: jest.fn(),
        entries: TESTPLAN_REPORT.entries,
        filter: null,
      }
      const createEntryComponent = jest.fn();
      const selectedUid = TESTPLAN_REPORT.uid;

      const navButtons = CreateNavButtons(
        props, createEntryComponent, selectedUid
      );
      expect(navButtons.length).toBe(props.entries.length);
      expect(navButtons).toMatchSnapshot();
    });
  });

  describe('GetSelectedUid', () => {
    it('gets the selected UID', () => {
      const selected = [TESTPLAN_REPORT];
      const uid = GetSelectedUid(selected);
      expect(uid).toBe(TESTPLAN_REPORT.uid);
    });
  });

});

