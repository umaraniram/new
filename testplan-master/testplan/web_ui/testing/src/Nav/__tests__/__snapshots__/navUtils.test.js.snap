// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`navUtils CreateNavButtons returns an array of nav buttons 1`] = `
Array [
  <ListGroupItem
    className="navButton_1x055n9-o_O-navButtonInteract_1ofjo98-o_O-unselectable_16maxrd"
    onClick={[Function]}
    tabIndex="1"
    tag="li"
  />,
  <ListGroupItem
    className="navButton_1x055n9-o_O-navButtonInteract_1ofjo98-o_O-unselectable_16maxrd"
    onClick={[Function]}
    tabIndex="2"
    tag="li"
  />,
]
`;
